### PR TITLE
Adds a key stretching algorithm and remove secret usefullness

### LINKED ISSUE #39: Must add a key stretching algorithm
To ensure password security, this plugin should feature a key stretching algorithm when hashing passwords to prevent short passwords to be easily breakable.

### PR DESCRIPTION
**Pull request by dbengsch**
:warning: Must be merged after #38

* Use a key stretching algorithm (pbkdf2)
* Each credential record holds its own salt; as a result `config.secret` becomes useless.

fixes #39

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1,132 +1,332 @@
...
   init (customConfig, context) {
-    let strategy;
     this.config = Object.assign(defaultConfig, customConfig);
 
     if (!this.config.algorithm) {
-      console.error(new Error('plugin-auth-passport-local: The \'algorithm\' attribute is required'));
-      return false;
+      return Promise.reject(new this.context.errors.BadRequestError(`plugin-auth-passport-local: The 'algorithm' attribute is required`));
+    }
+    if (['sha256', 'sha512'].indexOf(this.config.algorithm) === -1) {
```
Comment (benoitvidis) [inline lib/index.js]: nitpicking: why?
The following hash functions look perfectly acceptable to me: 
```
  'RSA-SHA224',
  'RSA-SHA256',
  'RSA-SHA384',
  'RSA-SHA512',
  'sha224',
  'sha224WithRSAEncryption',
  'sha256',
  'sha256WithRSAEncryption',
  'sha384',
  'sha384WithRSAEncryption',
  'sha512',
  'sha512WithRSAEncryption',

File: lib/index.js [hunk 2]
```diff
@@ -1,132 +1,332 @@
 const
-  isString = require('lodash.isstring'),
   crypto = require('crypto'),
-  pipes = require('./config/pipes'),
   PasswordManager = require('./passwordManager'),
-  Strategy = require('./passport/strategy'),
+  LocalStrategy = require('passport-local').Strategy,
   defaultConfig = {
-    'secret': null,
     'algorithm': 'sha256',
```
Comment (benoitvidis) [inline lib/index.js]: nitpicking: why prefer sha256 over more secure sha512?

File: lib/index.js [hunk 3]
```diff
@@ -1,132 +1,332 @@
...
+   * @returns {Promise<object>}
    */
-  encryptCredentials (request, callback) {
-    if (request.input.body.password === undefined) {
-      // The plugin does not enforce the password in case another plugin is installed
-      // @todo: add some configuration check on whether to trigger an error
-      return callback(null, request);
+  update (request, credentials, kuid) {
+    if (credentials.kuid) {
+      return Promise.reject(new this.context.errors.BadRequestError('The request must contain a kuid attribute.'));
```
Comment (xbill82) [inline lib/index.js]: This rejection message doesn't seem consistent with the condition

**Comment by xbill82** [comment]
I let you check the error message in my comment before merging.

File: lib/index.js (hunk 3 above)
Comment (dbengsch) [inline lib/index.js]: Merged underlying PR (was already corrected there)

**Comment by dbengsch** [comment]
@xbill82 @benoitvidis Done

**Pull request merged**